controllers/profile/profile.go

```diff
@@ -88,10 +108,64 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		return reconcile.Result{}, errors.Wrap(ignoreNotFound(err), errGetProfile)
 	}
 
+	for name, contents := range profile.Data {
+		profilePath, err := getProfilePath(name, profile)
+		if err != nil {
+			logger.Error(err, "cannot get profile path")
+			return reconcile.Result{}, err
+		}
+
+		if err = saveProfileOnDisk(profilePath, contents); err != nil {
+			logger.Error(err, "cannot save profile into disk")
+			return reconcile.Result{}, err
+		}
+	}
+
 	logger.Info("Reconciled profile", "resource version", profile.GetResourceVersion())
 	return reconcile.Result{RequeueAfter: longWait}, nil
 }
 
+func saveProfileOnDisk(fileName, contents string) error {
+	dirPath := path.Dir(fileName)
+	if _, err := os.Stat(dirPath); err != nil {
+		if os.IsPermission(err) {
+			return errors.Wrap(err, errCreatingOperatorDir)
+		}
+		if os.IsNotExist(err) {
+			if err := os.MkdirAll(dirPath, dirPermissionMode); err != nil {
+				return errors.Wrap(err, errSavingProfile)
+			}
+		}
+	}
+
+	if err := ioutil.WriteFile(fileName, []byte(contents), filePermissionMode); err != nil {
```

Comment:
Do we want to validate that the contents are actually a valid profile configuration?

Reply:
That is a very good point. I was already planning a follow-up PR with that.